cli/config_file.go

```diff
@@ -155,6 +162,91 @@ func SettingsFileReadWrite(state *Settings) *Settings {
 	return settings
 }
 
+// Check prerequisites
+func CheckRequirements() {
+	// getenv
+	envVarValue := os.Getenv("DOCKER_HOST")
+
+	if envVarValue == "" {
+		log.Fatalf("DOCKER_HOST environmental variable is empty or not set")
+	}
+
+	socketurl, err := url.ParseRequestURI(envVarValue)
+	if err != nil {
+		log.Fatalf("error: %v", err)
+	}
+
+	if socketurl.Host != "" {
+		log.Fatalf("DOCKER_HOST variable shouldn't have host")
+	}
+
+	if socketurl.Scheme != "unix" {
+		log.Fatalf("DOCKER_HOST variable should contain a valid unix socket")
+	}
+
+	// Check socket
+	cli, err := client.NewClientWithOpts(client.FromEnv, client.WithAPIVersionNegotiation())
+	if err != nil {
+		log.Fatalf("error: %v", err)
+	}
+
+	info, err := cli.Info(context.Background())
+	if err != nil {
+		log.Fatalf("error: %v", err)
+	}
+
+	switch info.InitBinary {
+	case "":
+		log.Printf("podman version: %s", info.ServerVersion)
+		PodmanInfo()
+	case "docker-init":
+		log.Printf("docker version: %s", info.ServerVersion)
+	default:
+		log.Fatalf("unknown init binary")
+	}
+}
+
+func PodmanInfo() {
+	cmd := exec.Command("podman", "info", "--format", "{{.Host.NetworkBackend}}")
+
+	stderr, err := cmd.StderrPipe()
+	if err != nil {
+		log.Fatalf("error: %v", err)
```

Comment:
`podman check error` would be better,  error is not really descriptive, here and docker checks too.

Reply:
Indeed, but it is also a good idea to append useful bits of information everywhere in error messages, since lot of places using this "format". Thanks!

Reply:
fixed, thanks!